### PR TITLE
Remove failing

### PR DESCRIPTION
**Pull request by PoignardAzur**

### CODE REVIEW

File: src/spacebro-client.js [hunk 1]
```diff
@@ -181,30 +182,29 @@ class SpacebroClient {
...
-      data._to = to
-      data._from = this.config.client.name
-      for (let pack of _filterHooks(eventName, this.packers)) {
-        data = pack({eventName, data}) || data
-      }
-      this.socket.emit(eventName, data)
-    } else {
-      this.logger.warn('can\'t emit, not connected.')
+    if (!this.connected) {
+      console.error(`Error: "${this.config.client.name}" is disconnected and cannot emit "${eventName}"`)
```
Comment (emmanuelgeoffray) [inline src/spacebro-client.js]: why did you not use this.logger ?

File: src/spacebro-client.js [hunk 2]
```diff
@@ -181,30 +182,29 @@ class SpacebroClient {
...
     this.events[eventName].add(handler, handlerContext, priority)
   }
 
   once (eventName, handler, handlerContext, priority) {
-    if (this.events[eventName]) {
-      this.logger.warn(`Signal ${eventName} already exists`)
+    if (!this.events[eventName]) {
+      this.events[eventName] = new Signal()
     }
     this.events[eventName] = new Signal()
```
Comment (emmanuelgeoffray) [inline src/spacebro-client.js]: is this line still useful ? the new Signal seems to be instantiated above.

File: src/spacebro-client.js (hunk 1 above)
Comment (PoignardAzur) [inline src/spacebro-client.js]: I interpreted this.logger as being intended for 'internal' logging (a message was received, connection started, etc); while "trying to emit from an unconnected socket" is a logic error that should never be silenced.

File: src/spacebro-client.js (hunk 2 above)
Comment (PoignardAzur) [inline src/spacebro-client.js]: No, you're right. It's an oversight from me; the tests didn't catch it.

File: src/spacebro-client.js (hunk 1 above)
Comment (emmanuelgeoffray) [inline src/spacebro-client.js]: 👍

File: test/tests-events.js [hunk 3]
```diff
@@ -65,15 +68,22 @@ test('once', async (t) => {
   await sleep(5000)
 })
 
-test.failing('on - Wildcard', async (t) => {
+test('on - Wildcard', async (t) => {
   const client = connect('emit-on-wildcard')
 
+  t.plan(5)
```
Comment (emmanuelgeoffray) [inline test/tests-events.js]: this fails, with 6 asserts.
It is weak to plan for a number of message received.
Instead, test exactly the 3 message are well received: hello, again, and...

Comment (PoignardAzur) [inline test/tests-events.js]: Done. I reworked this test.

**Comment by emmanuelgeoffray** [comment]
ok @PoignardAzur,
please merge master

**Comment by PoignardAzur** [comment]
Ready to merge

**Pull request merged**